osf/models/node.py

```diff
@@ -1189,6 +1189,10 @@ def add_contributor(self, contributor, permissions=None, visible=True,
         if contrib_to_add.is_disabled:
             raise ValidationValueError('Deactivated users cannot be added as contributors.')
 
+        if not contrib_to_add.is_registered and not contrib_to_add.unclaimed_records:
+            raise ValidationValueError('Unregistered users cannot be added as contributors without unclaimed records.'
```

Comment:
Not to be a broken record here, but I still think a potentially user facing error message using the term "unclaimed records" would be a bit confusing - How bout just something like "An error has occurred" or whatever other language we already use for a generic error on our end. The place where it was raised will be enough of a clue for devs to get the context.

Reply:
But this can be up to the discretion of the final reviewer, not a blocker. Other than that looks good!

Reply:
I agree with @erinspace -- "unclaimed records" is for internal use only. Can the error message be changed to something more understandable for external users?